pkg/process/checks/container_check_test.go

```diff
@@ -0,0 +1,139 @@
+
+package checks
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	proccontainersmocks "github.com/DataDog/datadog-agent/pkg/process/util/containers/mocks"
+)
+
+// Constant modeled after cacheValidityNoRT to allow for mocking containers functions
+const (
+	cacheValidityNoRTTest = 2 * time.Second
+)
+
+func containerCheckWithMockContainerProvider(t *testing.T) (*ContainerCheck, *proccontainersmocks.MockContainerProvider) {
+	mockCtrl := gomock.NewController(t)
+	containerProvider := proccontainersmocks.NewMockContainerProvider(mockCtrl)
+	sysInfo := &model.SystemInfo{
+		Cpus: []*model.CPUInfo{
+			{CoreId: "1"},
+			{CoreId: "2"},
+			{CoreId: "3"},
+			{CoreId: "4"},
+		},
+	}
+	hostInfo := &HostInfo{
+		SystemInfo: sysInfo,
+	}
+
+	return &ContainerCheck{
+		hostInfo:          hostInfo,
+		containerProvider: containerProvider,
+	}, containerProvider
+}
+
+func TestContainerCheckPayloads(t *testing.T) {
+	check, mockContainerProvider := containerCheckWithMockContainerProvider(t)
+	check.maxBatchSize = 2
+
+	// mock containers for testing
+	containers := []*model.Container{
+		{
+			Id: "container-1",
+			Addresses: []*model.ContainerAddr{
+				{Ip: "10.0.2.15", Port: 32769, Protocol: model.ConnectionType_tcp},
+				{Ip: "172.17.0.4", Port: 6379, Protocol: model.ConnectionType_tcp},
+			},
+		},
+		{
+			Id: "container-2",
+			Addresses: []*model.ContainerAddr{
+				{Ip: "172.17.0.2", Port: 80, Protocol: model.ConnectionType_tcp},
+			},
+		},
+		{
+			Id: "container-3",
+			Addresses: []*model.ContainerAddr{
+				{Ip: "172.17.0.3", Port: 88, Protocol: model.ConnectionType_tcp},
+			},
+		},
+	}
+
+	mockContainerProvider.EXPECT().GetContainers(cacheValidityNoRTTest, nil).Return(containers, nil, nil, nil)
+
+	// Test check runs without error using default chunk settings
+	result, err := check.Run(testGroupId(0), nil)
+	assert.NoError(t, err)
+
+	// Test that result has the proper number of chunks, and that those chunks are of the correct type
+	for _, elem := range result.Payloads() {
+		assert.IsType(t, &model.CollectorContainer{}, elem)
+		collectorContainers := elem.(*model.CollectorContainer)
+		resultContainers := collectorContainers.GetContainers()
+		for _, ctr := range resultContainers {
+			assert.Empty(t, ctr.Host)
+		}
+		if len(resultContainers) > check.maxBatchSize {
+			t.Errorf("Expected less than %d messages in chunk, got %d",
+				check.maxBatchSize, len(resultContainers))
+		}
+	}
+}
+
+func TestContainerCheckChunking(t *testing.T) {
+	for _, tc := range []struct {
+		noChunking            bool
+		expectedPayloadLength int
+	}{
+		{
+			noChunking:            false,
+			expectedPayloadLength: 2,
+		},
+		{
+			noChunking:            true,
+			expectedPayloadLength: 1,
+		},
+	} {
+		t.Run("NoChunking:"+strconv.FormatBool(tc.noChunking), func(t *testing.T) {
```

Comment:
[minor] This table test is great! It'll be easier to read if you just add a `name` attribute to your test struct. That way  you can give the test case and explicit name.